packages/core/src/bundle.ts

```diff
@@ -182,7 +187,13 @@ function makeBundleVisitor(version: OasMajorVersion, dereference: boolean, rootD
         if (!componentType) {
           replaceRef(node, resolved, ctx);
         } else {
-          if (dereference) {
+          if (resolved.node.title === ctx.key && componentType === 'schemas') {
```

Comment:
Using schema.title is not a good idea. What if you have two different schemas with the same title? There may be no title at all too.

You should be using inferred schema name: https://github.com/Redocly/openapi-cli/blob/02ab3766ad45987beb630c72324e8cb9a623a0bb/packages/core/src/bundle.ts#L243

Also, there is no constraint here that this schema is referenced within components section.

The original issue is about:

```
schemas:
  A:
    $ref: ./A.yaml
```

This code will also work in this case:

```
/operation:
  put:
    requestBody:
      application/json:
        schema:
          type: object
          properties:
            A:
              $ref: ./A.yaml
```

Reply:
> You should be using inferred schema name:

before I check whether schema should be placed inline I need to know original name of the reference so I cannot use `getComponentName` in this case. I came up with[ this solution](https://github.com/Redocly/openapi-cli/pull/439/files#diff-0bad609a2903ba228646e389eaf5a4de07f9e9917b9811df3814180433e88585R309)

> This code will also work in this case:
```yaml
/operation:
  put:
    requestBody:
      application/json:
        schema:
          type: object
          properties:
            A:
              $ref: ./A.yaml
```

It won't. We're checking the component type [here](https://github.com/Redocly/openapi-cli/pull/439/files#diff-0bad609a2903ba228646e389eaf5a4de07f9e9917b9811df3814180433e88585R316). Also we have a [test](https://github.com/Redocly/openapi-cli/pull/439/files#diff-a813ab9a158e235b8602a38c1fd939472bd18641d11166b53bb6a4083ae4e9e0R91) that covers this case

Reply:
It will though. The test you linked is not for schemas but my example is. Try my example.

Also, I see a few cases that won't work correctly. I'll add a few tests to this PR (failing) tomorrow morning.